lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -472,6 +500,14 @@ public AggregatedMemoryContext getMemoryContext()
         return memoryContext;
     }
 
+    public void validateWritePageChecksum(Page page)
```

Comment:
We need the page to be assembled from blocks so that we can cheaply iterate over `columnHashes` in `WriteChecksumBuilder#addPage`. But ParquetReader provides only `readBlock` and assembly of page happens in ParquetPageSource. That's why we feed page back to ParquetReader#validateWritePageChecksum from ParquetPageSource for validation.